From 474912aa4bf111785498ef8ea528192c859b17bb Mon Sep 17 00:00:00 2001
From: Pritam Srichandan Sahoo <PritamSrichandan.Sahoo@windriver.com>
Date: Thu, 20 Aug 2026 09:44:33 +0000
Subject: [PATCH 2/2] gpio: siul2-s32cc: fix RTC IRQ request failure on S32G2

Issue: LIN1025-16432

Add .map and .unmap callbacks to siul2_domain_ops pointing to the
exported gpiochip_irq_map() and gpiochip_irq_unmap() functions.

Without these callbacks, request_threaded_irq() returns -ENOSYS when
the pcf85063 RTC driver requests its GPIO-based interrupt on S32G2
boards, causing RTC alarms to be disabled:

rtc rtc0: unable to request IRQ, alarms disabled, err=-38

Upstream-Status: Backport [WRLinux LTS24 kernel v6.6/standard/nxp-sdk-6.6/nxp-s32g]
Signed-off-by: Pritam Srichandan Sahoo <PritamSrichandan.Sahoo@windriver.com>
---
 drivers/gpio/gpio-siul2-s32cc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpio/gpio-siul2-s32cc.c b/drivers/gpio/gpio-siul2-s32cc.c
index ef3e13ffd5f2..bfe264c4d228 100644
--- a/drivers/gpio/gpio-siul2-s32cc.c
+++ b/drivers/gpio/gpio-siul2-s32cc.c
@@ -931,6 +931,8 @@ static int siul2_irq_domain_xlate(struct irq_domain *d,
 }
 
 static const struct irq_domain_ops siul2_domain_ops = {
+	.map  = gpiochip_irq_map,
+	.unmap  = gpiochip_irq_unmap,
 	.xlate	= siul2_irq_domain_xlate,
 };
 
-- 
2.53.0

